iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -152,7 +152,7 @@ public override async Task CloseAsync(CancellationToken cancellationToken)
 
         #region Telemetry
 
-        public override async Task SendEventAsync(Message message, CancellationToken cancellationToken)
+        public override async Task SendEventAsync(MessageBase message, CancellationToken cancellationToken)
```

Comment:
The transport layer is updated to operation on `MessageBase`. This could be either `Message` or `TelemetryMessage`, depending on which public API calls it.